src/bp-core/admin/bp-core-admin-tools.php

```diff
@@ -803,6 +809,49 @@ function xprofile_update_display_names() {
 	}
 }
 
+/**
+ * Update nickname by removing invalid special characters.
+ *
+ * @since BuddyBoss x.x.x
+ */
+function bbp_repair_nicknames() {
+
+	$offset = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
+
+	$args = array(
+		'number' => 50,
+		'fields' => array( 'ID' ),
+		'offset' => $offset,
+	);
+
+	$users = get_users( $args );
+
+	if ( ! empty( $users ) ) {
+
+		foreach ( $users as $user ) {
+			$nickname = bp_get_user_meta( $user->ID, 'nickname', true );
+
+			if ( ! empty( $nickname ) ) {
+				$nickname = preg_replace( '/[^A-Za-z0-9-_\.]/', '-', $nickname );
+				bp_update_user_meta( $user->ID, 'nickname', $nickname );
+				$offset++;
```

Comment:
@ngolwalkar We should add the check here as well like the updated nickname should not exist.